kratos/processes/find_intersected_geometrical_objects_process.cpp

```diff
@@ -186,12 +187,11 @@ void FindIntersectedGeometricalObjectsProcess::Execute()
 
         const auto it_elements_begin = r_elements_array.begin();
 
-        #pragma omp parallel for private(leaves)
-        for (int i = 0; i < static_cast<int>(number_of_elements); i++) {
-            auto it_elem = it_elements_begin + i;
+        IndexPartition<std::size_t>(number_of_elements).for_each(leaves ,[&](std::size_t index, OtreeCellVectorType& local_leaves){
```

Comment:
Using first time, IndexPartition with TLS. Hope this is correct? @philbucher

Reply:
yes it is correct (although now technically it is the equivalent of `firstprivate` since the `leaves` are constructed beforehand

Hence we can simplify the code, see my suggestion (and remove the variable in line 181)

```suggestion
        IndexPartition<std::size_t>(number_of_elements).for_each(OtreeCellVectorType() ,[&](std::size_t index, OtreeCellVectorType& rLocalLeaves){
```